src/scripts/eastern_kingdoms/elwynn_forest/elwynn_forest.cpp

```diff
@@ -14,80 +14,149 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-/* ScriptData
-SDName: Elwynn_Forest
-SD%Complete: 50
-SDComment: Quest support: 1786
-SDCategory: Elwynn Forest
-EndScriptData */
-
-/* ContentData
-npc_henze_faulk
-EndContentData */
-
 #include "scriptPCH.h"
 
-/*######
-## npc_henze_faulk
-######*/
+namespace tome_of_divinity
+{
+    // Script name: tome_of_divinity
+    // Used by two npc's
+    // https://www.wowhead.com/classic/npc=6172/henze-faulk
+    // https://www.wowhead.com/classic/npc=6177/narm-faulk
 
-#define SAY_HEAL    -1000187
+    constexpr uint32 SAY_HEAL{ 2283 };
+    constexpr uint32 SPELL_SYMBOL_OF_LIFE{ 8593 };
 
-struct npc_henze_faulkAI : public ScriptedAI
-{
-    uint32 lifeTimer;
-    bool spellHit;
+    enum State
+    {
+        STATE_RESET = 0,
+        STATE_JUST_REVIVED,
+        STATE_DO_EMOTE
+    };
+}
 
-    npc_henze_faulkAI(Creature* pCreature) : ScriptedAI(pCreature)
+struct tome_of_divinityAI : public ScriptedAI
+{
+    explicit tome_of_divinityAI(Creature* pCreature) : ScriptedAI(pCreature)
     {
-        Reset();
+        tome_of_divinityAI::Reset();
     }
 
+    uint8 m_uiState{};
+
+    uint32 m_uiResetTimer{};
+    uint32 m_uiTalkTimer{};
+    uint32 m_uiEmoteTimer{};
+
+    ObjectGuid m_playerGuid{};
+
+    bool m_bSpellHit{};
+
     void Reset() override
     {
-        lifeTimer = 120000;
+        m_uiState = tome_of_divinity::STATE_RESET;
+
+        m_uiResetTimer = 120000;
+        m_uiTalkTimer = 1000;
+        m_uiEmoteTimer = 1000;
+
+        m_playerGuid.Clear();
+
+        m_bSpellHit = false;
+
         m_creature->SetUInt32Value(UNIT_DYNAMIC_FLAGS, UNIT_DYNFLAG_DEAD);
-        m_creature->SetStandState(UNIT_STAND_STATE_DEAD);   // lay down
-        spellHit = false;
+        m_creature->SetStandState(UNIT_STAND_STATE_DEAD);
+        m_creature->SetDeathState(JUST_DIED);
     }
 
-    void MoveInLineOfSight(Unit *who) override { }
-
-    void UpdateAI(uint32 const diff) override
+    void UpdateAI(const uint32 uiDiff) override
```

Comment:
We use east const in vmangos, why are you reverting this to west const? It goes against our chosen code style.